module_utils/foreman_helper.py

```diff
@@ -3,6 +3,46 @@
 
 import re
 import yaml
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    import ansible.module_utils.ansible_nailgun_cement
+    HAS_NAILGUN = True
+except ImportError:
+    HAS_NAILGUN = False
+    NAILGUN_IMP_ERR = traceback.format_exc()
+
+
+class ForemanAnsibleModule(AnsibleModule):
+
+    def __init__(self, argument_spec, **kwargs):
+        args = dict(
+            server_url=dict(required=True),
+            username=dict(required=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(type='bool', default=True),
+            state=dict(choices=['present', 'absent'], default='present'),
+        )
+        args.update(argument_spec)
+        super(ForemanAnsibleModule, self).__init__(argument_spec=args, **kwargs)
+
+        if not HAS_NAILGUN:
+            self.fail_json(msg='The nailgun Python module is required',
+                           exception=NAILGUN_IMP_ERR)
+
+
+def filter_module_params(module):
+    return {k: v for (k, v) in module.params.items() if v is not None}
+
+
+def get_server_params(module_params):
```

Comment:
I wonder if this should be changed into a method on a ForemanAnsibleModule instance `server, params = module.parse_params()`

Reply:
like this?